src/plugins/query_enhancements/public/query_assist/components/query_assist_summary.tsx

```diff
@@ -83,18 +89,23 @@ export const QueryAssistSummary: React.FC<QueryAssistSummaryProps> = (props) =>
 
   const sampleSize = 10;
 
-  const reportMetric = useCallback(
-    (metric: string) => {
-      if (props.usageCollection) {
-        props.usageCollection.reportUiStats(
-          METRIC_APP,
-          props.usageCollection.METRIC_TYPE.CLICK,
-          metric + '-' + uuidv4()
-        );
-      }
-    },
-    [props.usageCollection, METRIC_APP]
-  );
+  const queryChanged =
+    queryState.answer && queryState.answer !== props.data.query.queryString.getQuery().query;
+  // It can generate summary when
+  // 1. it has the current generated query(answer)
+  // 2. the generated query didn't been changed by user
+  // 3. there are search results
+  const canGenerateSummary = Boolean(results.length) && Boolean(queryState.answer) && !queryChanged;
+  // Generate summary can be auto triggered only when first time generating the query
+  const shouldAutoTrigger = !lastUsedQueryStateRef.current;
+  // Display a message in the panel to indicate that clicking
+  // "Generate summary" button is needed to generate the summary
+  const manualTriggerVisible =
+    !shouldAutoTrigger &&
+    canGenerateSummary &&
+    lastUsedQueryStateRef.current?.answer !== queryState.answer;
```

Comment:
should we compare result as well?

Reply:
Maybe unnecessary and too much to have a deep equality check on the results? The "Generate summary" button will be available to for a manual trigger if needed when results updated.